### ISSUE TITLE
YOLOv8 custom training issue: Model misclassifies random objects with high confidence

### ISSUE THREAD

**Issue by ritageelsayed350-byte** [ultralytics/ultralytics#22826]
It's my first time to train AI models, and I'm using YOLOv8 and trained a custom model on my own dataset (cars only).
The model was working normally, but suddenly it started detecting random objects as cars — things like my face, walls, or empty background, and sometimes with very high confidence. How can i solve this problem?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ritageelsayed350-byte, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get support as quickly as possible. An Ultralytics engineer will also review your issue and assist you soon 😊

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Since your report is about custom training and unexpected detections, it sounds like a custom training ❓ Question, but if you believe this is a framework issue rather than a dataset/training configuration problem, please also treat it as a potential 🐛 Bug Report.

-If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.  
-If this is a custom training ❓ Question, please provide as much information as possible, including:
- A few example images from your dataset (with labels visible if possible)
- Your training command and full logs
- Model you used (e.g., YOLOv8n, YOLOv8s, etc.)
- Validation metrics (precision/recall, mAP) and how they changed over time
- Any post-processing or custom code you are using

Also, please verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/), as many issues with overconfident false positives or unstable predictions can be traced back to data quality, labeling, or training configuration details.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Before we investigate further, please upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a **Python>=3.8** environment with **PyTorch>=1.8** to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

Also let us know:
- Your `ultralytics` version (`yolo checks` output)
- Your OS and hardware (CPU/GPU)
- How you are running training/inference (Python script, CLI, notebook, etc.)

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Please update your issue with the details requested above so the engineering team can better understand what’s happening and help you more effectively 🙌

**Comment by Y-T-G** [comment]
Your dataset is probably small and not diverse which caused the model to overfit. You need to increase the diversity of dataset. Add random images without any labels.
